### ISSUE TITLE
form_template override on struct block can not accept Jinja2 html

### ISSUE THREAD

**Issue by bmoe872** [wagtail/wagtail#5686]
### Issue Summary

When overriding the `form_template` on a custom structblock you can not use Jinja2 templates to override the template. The admin renders the template as a string, as opposed to as HTML. I would have expected it to render the HTML, or I would hope for something in the documentation warning about this.

### Steps to Reproduce

1. Define a custom struct block
2. Use:
```
class Meta:
    form_template = "<path to jinja2 template>"
```
3. The Jinja2 template can look like this:
```
<div class="{{ classname }}">
{% for key, child in children.items() %}
    {{ child.render_form() }}
{% endfor %}
</div>
```
4. Go to admin where you entered this custom structblock, and it will render out the HTML as a string with `"`'s wrapped around it. 

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: `3.6.8`
* Django version: `2.2.7`
* Wagtail version: `2.7.0`
* Browser version: `Chrome Version 78.0.3904.87 (Official Build) (64-bit)`
* Jinja2 version: `2.10.1`

**Comment by bmoe872** [comment]
For now I've found a workaround by overriding the `render_form` function on the structblock, but this feels hacky.

```
from django.utils.safestring import mark_safe

    def render_form(self, value, prefix='', errors=None):
        context = self.get_form_context(value, prefix=prefix, errors=errors)

        return mark_safe(render_to_string(self.meta.form_template, context))
```

**Comment by gasman** [comment]
Hi @bmoe872 - thanks for the report.

I think there's a reasonable argument that jinja2 is at fault here, for not returning a SafeString like Django's own template engine does. However, I don't think there's any harm in us correcting for that by adding the `mark_safe`, as that will have no effect on a template engine that does return a SafeString. Happy to accept a PR that adds this.

**Comment by bmoe872** [comment]
@gasman you got it, PR'd!

**Comment by gasman** [comment]
Fixed in #5687.
